util.c

```diff
@@ -783,7 +783,86 @@ size_t filter_result_len(const char *filter, const char *user, char *output) {
 }
 
 char *filter_printf(const char *filter, const char *user) {
-  char *result = malloc(filter_result_len(filter, user, NULL));
-  filter_result_len(filter, user, result);
+  char *user_rfc4515 = rfc4515_replace(user);
+  char *result = malloc(filter_result_len(filter, user_rfc4515, NULL));
+  filter_result_len(filter, user_rfc4515, result);
+  free(user_rfc4515);
   return result;
 }
+
+size_t rfc4515_length (const char* in, char* out)
+{
+    const char* pos = NULL;
+    size_t result = 0;
+    const char specials[] = "\\*()";
+
+    do
+    {
+        pos = strpbrk(in, specials);
+
+        size_t len = 0;
+        if(pos != NULL)
+        {
+            len = pos - in;
+        }
+        else
+        {
+            len = strlen(in);
+        }
+
+        if(out != NULL)
+        {
+            strncpy(out, in, len);
+            out += len;
+        }
+
+        result += len;
+        in += len;
+
+        if(pos != NULL)
+        {
+            if(out != NULL)
+            {
+                *out++ = 0x5c;
+
+                if( *pos == 0x5c )
```

Comment:
When the code expands like this it might be better to do something different. I think collapsing all of this into a snprintf call might be better. something like: `snprintf(out, 3, "%c%02x", 0x5c, *pos)` that might need additional error handling, but is better than repeating if-else statements.

Reply:
Hey @klali, `pos` can't be NULL and is always one of the special chars. And `out` can't be NULL either. Did I miss something?

Reply:
in regards to error handling or in regards of snprintf/sprintf ?

since everything is known sprintf() is probably fine, it should always write 3 bytes (+ NULL) and it should always succeed.

Reply:
Hey @klali, in regards to error handling. I optimized my code.

Reply:
maybe that should've been `snprintf(out, 4, "\%02x", *pos)` but nevermind, same effect =)